ea-2021-04-ndvi-automation-nkorinek.py

```diff
@@ -0,0 +1,646 @@
+# 
+# * San Joaquin Experimental Range (SJER) in Southern California, United States
+# * Harvard Forest (HARV) in the Northeastern United States
+# 
+# The data that you will use for this week is available from **earthpy** using the following download: 
+# 
+# `et.data.get_data('ndvi-automation')`
+# 
+# ## Assignment Goals
+# 
+# Your goal in this assignment is to create the most efficient and concise workflow that you can that allows for:
+# 
+# 1. The code to scale if you added new sites or more time periods to the analysis.
+# 2. Someone else to understand your workflow.
+# 3. The LEAST and most efficient (i.e. runs fast, minimize repetition) amount of code that completes the task.
+# 
+# ### HINTS
+# 
+# * Remove values outside of the landsat valid range of values as specified in the metadata, as needed.
+# * Keep any output files SEPARATE FROM input files. Outputs should be created in an outputs directory that is created in the code (if needed) and/or tested for.
+# * Use the functions that we demonstrated during class to make your workflow more efficient.
+# * BONUS - if you  chose - you can export your data as a csv file. You will get bonus points for doing this.
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+
+# YOUR CODE HERE
+import os
+from glob import glob
+#import functools
```

Comment:
@nkorinek  looks like you have a unneeded import in your code that is commented out.